torch/csrc/utils/six.h

```diff
@@ -25,15 +33,25 @@ inline bool isTuple(PyObject* obj) {
   return isTuple(pybind11::handle(obj));
 }
 
-inline PyObject *toTuple(PyStructSequence *obj) {
-  // create a new tuple object on python 2, or increase
-  // the ref count of the current object on python 3.
+// maybeAsTuple: if the input is a structseq, then convert it to a tuple
+//
+// On Python 3, structseq is a subtype of tuple, so these APIs could be used directly.
+// But on Python 2, structseq is not a subtype of tuple, so we need to manually create a
+// new tuple object from structseq.
+inline THPObjectPtr maybeAsTuple(PyStructSequence *obj) {
```

Comment:
returned_structseq_repr(PyStructSequence *obj) will now fail to compile

Reply:
@VitalyFedyunin fixed